app/controllers/sign_up_sheet_controller.rb

```diff
@@ -150,49 +150,59 @@ def redirect_to_assignment_edit(assignment_id)
     redirect_to controller: 'assignments', action: 'edit', id: assignment_id
   end
 
+  def find_signed_up_topics student_bids
+      signed_up_topics = []
+      student_bids.each do |bid|
+        sign_up_topic = SignUpTopic.find_by(id: bid.topic_id)
+        signed_up_topics << sign_up_topic if sign_up_topic
+      end
+     # signed_up_topics &= sign_up_topics
+      return signed_up_topics
+  end
+
+  def find_selected_topics assignment
+
+    # Find whether the user has signed up for any topics; if so the user won't be able to
+    # sign up again unless the former was a waitlisted topic
+    # if team assignment, then team id needs to be passed as parameter else the user's id
+    users_team = SignedUpTeam.find_team_users(assignment.id, session[:user].id)
+    selected_topics = if users_team.empty?
+                         nil
+                       else
+                         # TODO: fix this; cant use 0
+                         SignedUpTeam.find_user_signup_topics(assignment.id, users_team.first.t_id)
+                       end
+    return selected_topics
+  end
+
   def list
     @participant = AssignmentParticipant.find(params[:id].to_i)
     @assignment = @participant.assignment
     @slots_filled = SignUpTopic.find_slots_filled(@assignment.id)
     @slots_waitlisted = SignUpTopic.find_slots_waitlisted(@assignment.id)
     @show_actions = true
     @priority = 0
-    @sign_up_topics = SignUpTopic.where(assignment_id: @assignment.id, private_to: nil)
     @max_team_size = @assignment.max_team_size
     team_id = @participant.team.try(:id)
+    @student_bids = team_id.nil? ? [] : Bid.where(team_id: team_id).order(:priority)
```

Comment:
I am not sure why you add `.order(:priority)`. But this addition seems reasonable.